Vizzuality/heco-invest
# Project developers can apply to an open call

## Description

**In this PR:**  
- Added a new `useApplyToOpenCall` service/hook to `open-call-service`  
  _used to submit a new open call application_  
- Added `OpenCallApplication` types and form validation schema  
- Added a new `OpenCallApplicationModal` component  
  _modal with the form where the PD can choose a project & message to apply to an open call_   
- OpenCall page changes
  - A PD can apply to an OpenCall from both the header and the CTA at the footer  
    _Using the new `OpenCallApplicationModal`_
  - The open call application buttons are disabled for:
    - Investors  
    - Non-signed in users  
    - Open calls with a status other than `Launched`

**Others:**  
- I've added a new prop to the `Combobox` component so we can have a smaller listbox  
  _This was needed because the default size would not fit the `OpenCallApplication` modal_

## Testing instructions

Verify that:  
- Using a PD account:  
  - Navigate to a closed OpenCall page
    - Verify that the user cannot apply to the open call  
  - Navigate to an open OpelCall page  
    - Verify that the user can open the open call application modal  
    - The form validation works  
      _we don't have a way to view open call applications yet, so the easiest way to confirm it is submitted successfully is to check if the request goes through via devtools/network. We can also grab the `id` and verify that it can be deleted via swagger_  
    - Attempting to re-apply to the open call results in an "open call already taken" error  
      _To test that the modal/form can display errors coming from the endpoint_  
- Using an Investor account  
  - Verify that the user cannot apply to an open call
    _the buttons in the open call page are disabled_
- Without being signed in  
  - Verify that the user cannot apply to an open call
    _the buttons in the open call page are disabled_

## Tracking

[LET-898](https://vizzuality.atlassian.net/browse/LET-898)
[LET-983](https://vizzuality.atlassian.net/browse/LET-983)

## Screenshots

**Modal/form**  
<img width="591" alt="Screenshot 2022-08-29 at 12 57 29" src="https://user-images.githubusercontent.com/6273795/187195940-c6bd342b-c589-4a63-9242-00f058a19692.png">



**FE validation errors**  

<img width="591" alt="Screenshot 2022-08-29 at 12 57 56" src="https://user-images.githubusercontent.com/6273795/187196024-4eb65de7-9c32-4168-aedd-3ebc087a6868.png">

**BE alerts**  

<img width="588" alt="Screenshot 2022-08-29 at 12 58 36" src="https://user-images.githubusercontent.com/6273795/187196168-54f858e3-3629-4c6e-9783-1fec716291ff.png">
